### ISSUE TITLE
Discussion: Drop Bitcoin support completely

### ISSUE THREAD

**Issue by rappo** [bountysource/core#1086]
Opening this up to discussion from the community.

I personally want to drop Bitcoin support. BTC adds more administrative overhead than I feel is worth and it's a major pain in my ass:

* 100% of fraudulent cashout requests use BTC as cashout method
* Coinbase transactions time out after a few minutes, resulting in frequent mispayments that someone (always me) needs to refund to the customer. (and Coinbase as a system is also a pain to deal with.)
* BTC makes up a tiny fraction of our revenue, both in volume and in totals (no one likes to spend BTC...).  We'll have dozens of Paypal transactions a day, but *maybe* one BTC transaction every few days.
* One less integration / failure point if we get rid of it.

...Now that I actually typed this all out, I've started to solidify my opinion to drop bitcoin support.

**Comment by jamesweech** [comment]
I know the payments via it are little and it sounds like the implementation via coinbase could be worked upon but in my opinion it is becoming more important day to day and is more useful where other payment methods might be restricted.

**Comment by bparker06** [comment]
What about the people who either choose not to use PayPal, or have had their PayPal accounts shutdown for bogus reasons (they are allowed to shut it down for whatever reason they want) and have no other option?

**Comment by kode54** [comment]
Clearly they need to support wire transfers directly. Those are never done fraudulently. Clearly.

**Comment by twinaphex** [comment]
We were thinking of using BountySource for helping accelerate development for RetroArch and libretro and this decision puts us in a serious bind now to the point where we might have to research other alternatives.

We have had Paypal previously shut us down for arbitrary nonsensical reasons like 'selling emulators'. Everybody who has ever known us these past 7 years knows that if there is one thing we have been consistent on, is that we have never, ever sold an 'emulator' whatsoever, and even if we were, I fail to see how this would be any kind of 'terms of use' violation. So basically, they freeze your entire account for half a year, and you are not allowed to appeal their decision even though they are in the wrong.

So if we were to have to use Paypal only from this point on, we would have to put up with this kind of nonsense and unpredictability, to the point where you can't really rely on it because it's just so arbitrary and unreasonable.

**Comment by RogWilco** [comment]
I'd encourage you to reconsider. I realize the cost/benefit may seem a bit off balance right now, but I think it would be a mistake to drop Bitcoin support in the greater scheme of things. For now it may be largely symbolic, but consider that Bitcoin is also the universal alternative for people that cannot use your other accepted payment methods. That's going to be a bigger deal as Bountysource's global audience grows.

**Comment by Riiume** [comment]
Please keep BTC support! I have some suggestions to relieve the overhead:

2) Coinbase issues - There are [other BTC payment processors](https://en.bitcoin.it/wiki/How_to_accept_Bitcoin,_for_small_businesses#Merchant_Services) who might have better handling (or avoid entirely) these timeout situations.

3) BTC is lower volume - I foresee that changing, especially for purposes of paying devs in places not using the USD or Euro (e.g. India). The number of BTC related projects on GitHub is steadily increasing, so with patience I think the volume in BTC on BountySource will increase.

3.a) Why give up the niche of Bitcoin dev bounties to a competitor? Keeping this feature prevents your competitors from one-upping you by adding that feature.

4) Simpler is attractive but not necessarily-- paper spreadsheets are simpler than computers running Lotus or Excel, but the companies who adopted the computer ultimately out-competed their simplicity-favoring peers.
Bitcoin might be more complex to use at this point in time, but as the UI's and customer-facing software matures, it will become as easy as using any other Windows application.

**Comment by danielfl** [comment]
I'd also encourage you to reconsider The bitcoin payment is the only possible payment option for 4 billion debanked people around the the world. Even if this option didn't get the proper traction in bountysource yet please keep this in mind.

**Comment by Riiume** [comment]
There are many Bitcoin devs who would be happy to volunteer their services to maintain your Bitcoin FFI's and API's, and solve the bugs you may have!

Reach out to places where Bitcoin developers congregate and explain the issues you're facing and I guarantee you'll attract talented volunteers ready to work hard to keep you in the fold!

**Comment by rappo** [comment]
@jamesweech, @bparker06, @RogWilco, @Riiume, @danielfl, -- Thank you all for your feedback

@twinaphex I'm a *huge* fan of RetroArch / libretro. <3. Personally, I'd love to see you guys on the platform.

@kode54 Keep your comments constructive. I appreciate sarcasm as much as the next guy, but you added nothing to the discussion.

Here's what removing BTC would mean:

### Incoming payments

Bounties would be funded either by Paypal, or by a team/personal balance. Your Team/personal balance is funded either by earning bounties, or from Salt, our monthly recurring funding platform. Salt accepts Paypal or Credit/Debit Card (via Stripe).

That is to say, if your project has monthly donations via Salt, it's very likely that many of them avoid Paypal completely by using credit/debit cards. Your team can fund bounties with this Salt money without ever touching Paypal.

#### Invoicing for large orders

We have been lowkey about this, but for large order (usually $10k or more) we invoice directly with companies. You may have noticed how active [IBM](https://www.bountysource.com/teams/ibm) is... $100k in open bounties, $240k paid

#### Credit Card, ACH support

Adding Stripe support (CC, ACH) directly to bounty transactions is something the Bountysource team supports.

### Outgoing payments

Paypal is, by far, our most popular cash out method. We also process cashouts by check and, by request, bank wire.

Not all BTC cashouts are fraud, but all of our fraudulent cashouts are BTC. 

### Lost revenue from BTC

The lost revenue from dropping BTC is so small that it would not be noticed. Like, laughably small. Everyone who rallied for us to add Bitcoin support should be embarrassed at how little utility our marketplace got from it. We added bitcoin support ~3 years ago and it has made no noticeable impact. 

The simple fact is that those who have BTC are holding it as an investment and not spending it. If people naturally spent BTC in a way that made a difference, I would have never had to make this post in the first place. 

Having BTC support sounds cool, and I personally still believe that BTC will be very important in the future, but it is not a net-positive for Bountysource.

### What about the longer term vision for Bitcoin...

If it makes sense to add BTC support in the future, that'll be a relatively simple task. We've already done all the integration work with Coinbase (and could in theory do it with Stripe instead), so adding it back would be minor technical and bookkeeping work.

**Comment by kode54** [comment]
They're not talking about PayPal issues from accepting payments, they're talking about cashing out with PayPal, and getting shut down for handling so much money with next to no explanation. Receiving money with PayPal is always a dangerous prospect, because you always have to be quick to pull it out of the PayPal balance, just in case it's that special day where PayPal decides to lock your account.

**Comment by barkovv** [comment]
@rappo 
PayPal splits countries to "only send" and "send & receive" ones. It's okay when you leave in USA or Europe but is terrible since I am living in Russia. There is not legal & safe way I can get money via PayPal since Russia is "only send" country. I wish I could receive cash outs via PayPal but I can't. So if you drop BTC support completely I wish there would be some other that BTC way I can get the money. 

Is BTC okay if someone or me implement #1080 ?
Is BTC okay if bountysource switches for CoinBase to platform which is not so awful to use?

**Comment by rappo** [comment]
Again, thank you all for allowing me to vent and for responding.

I realize BTC has great value to those who can't or won't use Paypal etc. Open-source is a global community and it's important that we don't abandon people due to their region. That said, Bountysource is run on a volunteer basis in our spare time ...and I need to keep my sanity.

## Due to your feedback, we won't be ditching BTC. But we as a community need to solve these problems ASAP!

Here are the issues with BTC that would need to be solved, and I would need help from everyone to solve them, if we are to continue using BTC (in addition to workflow improvements like #1079 and #1080). This also ignores the fact that **no one is spending their damn Bitcoin**.

### Fraud

(and this really is our problem, @kode54) Here's how it happens. Someone creates a few bogus accounts and a bogus team. They use a stolen / hacked Paypal account to fund said team. They immediately request a BTC cashout.

#### How this affects Bountysource

1. Identify fraud. There's usually a clear pattern -- some obviously bullshit team, usually a day old user account, @gmx.de email, overly-generic cash out addresses, etc
2. Go into Paypal, refund suspected fraud transactions (or respond to the open case, if it's already been reported)
3. Cancel Cash out
4. Delete User(s)
5. Delete Team(s)

This also leaves a deleted team with a >$0 balance, which needs to be zero'd out. This whole thing is tedious and it's a drain on me when I just want to go in and process some cash outs for you guys.

### Mispayments through Coinbase

If you check out via BTC and you're paying from an external (non-Coinbase) wallet, you have some pre-determined amount of time (5 minutes?) for your transaction to be noticed, or Coinbase will reset. If you sent your transaction and it was reset, it'll show up as mispaid in Coinbase, which doesn't trigger a Bountysource event, and will never process your shopping cart.

#### How this affects Bountysource

1. User it "too slow" to pay and notices the cart never clears, no bounty/etc created
2. User emails support
3. I ask for return address, and wait for their reply
4. Refund BTC, ask them to try again
5. This is further complicated due to BTC volatility. What if this process takes 2-3 weeks to resolve and 1BTC went from $800 to $1200? Do I sent them $1200 or $800?

**Comment by kode54** [comment]
Sorry, I did not realize that people were using BountySource to launder stolen PayPal accounts.

At the same time, I too am annoyed by Bitcoin services having transaction timers, especially considering that it takes a bare minimum of 10 minutes for a block to be processed, and there's no guarantee that an immediate transaction will even make it into the current or even next block, considering how much traffic is being wasted on stupid sites that irresponsibly run thousands of microtransactions against the block chain in real time instead of using internal accounting. Stupid dice crap.

**Comment by frnco** [comment]
There are ways to fix fraudulent cash out requests as well as coinbase issues, but the overhead for a small fraction of the community it's probably not worth it.

Considering the number of critical issues and the fact that this seems to be very low-priority, I don't think BountySource can currently afford to spend Dev time on it.

Setting up a way for people to sponsor this feature, on the other hand, could be a good compromise. At the very least it shifts the responsibility over this decision to the entire community and avoid people complaining about it.

**Comment by lukespragg** [comment]
Perhaps only enable Bitcoin as an option for select accounts after X criteria is met? I'm not sure what the criteria would entail, but I'm sure some trusted users would appreciate it. I can't imagine that it'd be too much a burden then for the handful that do use it legitimately.

**Comment by rappo** [comment]
I've spun the fraud topic off into it's own Issue: https://github.com/bountysource/core/issues/1088
I'll do the same with the Coinbase mispayment issue, but I don't have time at the moment.

It would be great if people can contribute to solving #1088. And we can certainly use a developer to step up and help with it as well.

@frnco -- are you suggesting that BTC is not worth the effort and we (at least temporarily) stop supporting it?

**Comment by frnco** [comment]
@rappo I just pointed out resources are limited and must be allocated according to priorities. If BTC turns out to be a priority, be it because it attracts finding or because the core team considers it a must-have, then it should be treated as a priority, but if it is just a superfluous feature and it is actually negatively impacting maintenance of the features that are actually the priorities, it should be dropped.

Ultimately this is a decision for you and Warren, unless maybe if the community actually steps up and takes ownership of that feature, making it not affect your priorities negatively and, thus, painless to support it. That's why I suggested allowing people to fund it.
